Wikipedia/Code/WMFImageController.swift

```diff
@@ -411,54 +390,39 @@ extension WMFImageController {
      - returns: `AnyPromise` which resolves to `WMFImageDownload`.
      */
     @objc public func fetchImageWithURL(url: NSURL?) -> AnyPromise {
-        return AnyPromise(bound: fetchImageWithURL(url))
-    }
-    
-    /**
-     Objective-C-compatible variant of fetchImageWithURL(url:options:) returning an `AnyPromise`.
-     
-     - returns: `AnyPromise` which resolves to `WMFImageDownload`.
-     */
-    @objc public func fetchImageWithURL(url: NSURL?, options: SDWebImageOptions) -> AnyPromise {
-        return AnyPromise(bound: fetchImageWithURL(url, options: options))
+        return AnyPromise(bound: Promise<WMFImageDownload> { fulfill, reject in
```

Comment:
@coreyfloyd yeah this is step 1 - wrap the new stuff in promise kit for obj-c. I did this to get it to build so that Wikimedia-swift would update and I could use the new Obj-c compatible versions